docs/example/skip-constraints-example.md

```diff
@@ -0,0 +1,58 @@
+# Example: Building Collections with Conflicting Versions
+
+This example demonstrates how to use the `--skip-constraints` option to build wheel collections containing conflicting package versions.
+
+## Use Case
+
+Suppose you need to build a package index that contains multiple versions of the same package for different downstream consumers. For example, you might want to include both Django 3.2 and Django 4.0 in your collection.
+
+## Requirements Files
+
+Create a requirements file with conflicting versions:
+
+### requirements-conflicting.txt
+
+```text
+django==3.2.0
```

Comment:
I documented an example with Django, but I'm happy to do it for another package if more appropriate.